SCClassLibrary/Common/Audio/Splay.sc

```diff
@@ -1,18 +1,33 @@
+LevelComp {
+	*new { |levelComp, rate, n|
+		if (levelComp == true) {
+			^if(rate == \audio) {
+				// ar default is equal power for backwards compatibility
+				n.reciprocal.sqrt
+			} {
+				// kr default is equal amplitude
+				n.reciprocal
+			}
+		};
+		// if false, level is untouched
+		if (levelComp == false) { ^1 };
+
+		// now levelComp is a number or UGen,
+		// so we scale by exponent: 0 is no change,
+		// 0.5 is square root = equal power
+		// 1.0 is level / numchans = equal amplitude
+		^n.reciprocal ** levelComp.clip(0.0, 1.0)
+	}
+}
+
 Splay : UGen {
 
 	*new1 { arg rate, spread = 1, level = 1, center = 0.0, levelComp = true ... inArray;
 
 		var n = max(2, inArray.size);
 		var n1 = n - 1;
 		var positions = ((0 .. n1) * (2 / n1) - 1) * spread + center;
-
-		if (levelComp) {
-			if(rate == \audio) {
-				level = level * n.reciprocal.sqrt
-			} {
-				level = level / n
-			}
-		};
+		level = level * LevelComp(levelComp, rate, n);
```

Comment:
(since we talked about multichannel expansion: this will expand fine, because it is called in `*new1`, which is already expanded.)

So all good!

Reply:
great, thanks for testing!